snap/snapcraft.yaml

```diff
@@ -68,72 +72,94 @@ parts:
         - libsodium23
         - libxml2
         - libxslt1.1
+        - libpython3.8-minimal
+        - libpython3.8-stdlib
+        - python3-distutils
+        - python3-pip
+        - python3-setuptools
+        - python3.8
         - squashfs-tools
         - xdelta3
+    build-environment:
+        - "CRYPTOGRAPHY_DONT_BUILD_RUST": "1"
     override-build: |
-      snapcraftctl build
+        snapcraftctl build
+
+        echo "Create libsodium symlink..."
+        TRIPLET_PATH="$SNAPCRAFT_PART_INSTALL/usr/lib/$(gcc -print-multiarch)"
+        LIBSODIUM="$(readlink -n "$TRIPLET_PATH/libsodium.so.23")"
+        ln -sf "$LIBSODIUM" "$TRIPLET_PATH/libsodium.so"
+
+        # Default to python3.8.
+        ln -sf python3.8 "$SNAPCRAFT_PART_INSTALL/usr/bin/python3"
+
+        # Install requisite python packages to dist-packages, which will be picked
```

Comment:
Can you clarify why this is dist-packages and not site-packages?

Reply:
Because dist-packages is automatically picked up by the interpreter when using $SNAPCRAFT_STAGE/usr/bin/python3 whereas a site-packages is not.

Reply:
I realize now it may be related sitecustomize, I jut force pushed a diff to handle this for the legacy case:
https://github.com/snapcore/snapcraft/compare/14fa5a445de15a0d3bf5c9cf275117532d63ea6f..598f330fac566e290bdbb5d412f7a32906fdf7f1

Reply:
I dropped that by installing (a copy of) sitecustomize.py prior to those installs...